### PR TITLE
add subscribtion controller and router

### LINKED ISSUE #59: add subscription
add subscription controllers and router

### PR DESCRIPTION
**Pull request by heleena-fadi**
Relates #59
Closes #59

### CODE REVIEW

**Review by HamdyPs — APPROVED** [review]

File: src/controllers/subscriptions/addSubscription.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+const createSubscription = async (
+  req: UserRequestI,
+  res: Response,
+  next: NextFunction,
+
+): Promise<void> => {
+  try {
+    const { id } = req.user;
+    const { classId } = req.params;
+    const { price, status } = req.body;
```
Comment (HamdyPs) [inline src/controllers/subscriptions/addSubscription.ts]: we won't get the status from client

File: src/database/subscriptionSchema.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'users',
+  },
+
+  classId: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'classes',
+  },
+
+  price: {
```
Comment (HamdyPs) [inline src/database/subscriptionSchema.ts]: the price of the class must be in the class itself

File: src/interfaces/subscription.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { Types } from 'mongoose';
+
+interface ISubscription {
+    userId: Types.ObjectId
+    classId: Number
+    price: Number
```
Comment (HamdyPs) [inline src/interfaces/subscription.ts]: no need for price anymore

File: src/routers/subscription.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import express, { type Router } from 'express';
+
+import addSubscription from '../controllers/subscriptions';
+
+const SubscriptionRouter: Router = express.Router();
+
+SubscriptionRouter.post('/:userId', addSubscription);
```
Comment (HamdyPs) [inline src/routers/subscription.ts]: this isn't user id. it's a class id

Comment (HamdyPs) [inline src/routers/subscription.ts]: check if the user signed in

**Review by HamdyPs — CHANGES_REQUESTED** [review]

**Review by HamdyPs — APPROVED** [review]

**Review by AdhamHaisami — APPROVED** [review]

**Pull request merged**